cli/module_loader.rs

```diff
@@ -21,13 +47,210 @@ use deno_core::ModuleType;
 use deno_core::OpState;
 use deno_core::ResolutionKind;
 use deno_core::SourceMapGetter;
+use deno_graph::source::Resolver;
 use deno_graph::EsmModule;
 use deno_graph::JsonModule;
+use deno_graph::Module;
+use deno_graph::Resolution;
+use deno_lockfile::Lockfile;
+use deno_runtime::deno_node::NodeResolutionMode;
 use deno_runtime::permissions::PermissionsContainer;
+use deno_semver::npm::NpmPackageReqReference;
+use std::borrow::Cow;
 use std::cell::RefCell;
+use std::collections::HashSet;
 use std::pin::Pin;
 use std::rc::Rc;
 use std::str;
+use std::sync::Arc;
+
+pub struct ModuleLoadPreparer {
+  options: Arc<CliOptions>,
+  caches: Arc<Caches>,
+  deno_dir: DenoDir,
+  graph_container: Arc<ModuleGraphContainer>,
+  lockfile: Option<Arc<Mutex<Lockfile>>>,
+  maybe_file_watcher_reporter: Option<FileWatcherReporter>,
+  module_graph_builder: Arc<ModuleGraphBuilder>,
+  npm_resolver: Arc<NpmPackageResolver>,
+  parsed_source_cache: Arc<ParsedSourceCache>,
+  progress_bar: ProgressBar,
+  resolver: Arc<CliGraphResolver>,
+}
+
+impl ModuleLoadPreparer {
+  #[allow(clippy::too_many_arguments)]
+  pub fn new(
+    options: Arc<CliOptions>,
+    caches: Arc<Caches>,
+    deno_dir: DenoDir,
+    graph_container: Arc<ModuleGraphContainer>,
+    lockfile: Option<Arc<Mutex<Lockfile>>>,
+    maybe_file_watcher_reporter: Option<FileWatcherReporter>,
+    module_graph_builder: Arc<ModuleGraphBuilder>,
+    npm_resolver: Arc<NpmPackageResolver>,
+    parsed_source_cache: Arc<ParsedSourceCache>,
+    progress_bar: ProgressBar,
+    resolver: Arc<CliGraphResolver>,
+  ) -> Self {
```

Comment:
This is gross... Maybe we could have an option struct for this purpose in the future?

Reply:
I think it's ok because this is just for initializing the dependencies, so an options struct wouldn't help much with making this explanatory. I'm going to slowly whittle down the number here by extracting out more resonsibilities (ex. creating a type checker struct will eliminate two and replace with one)